akka-typed/src/main/scala/akka/typed/persistence/scaladsl/PersistentActor.scala

```diff
@@ -41,19 +41,26 @@ object PersistentActor {
    * Factories for effects - how a persitent actor reacts on a command
    */
   object Effect {
-    def persist[Event, State](event: Event): Effect[Event, State] =
-      new Persist[Event, State](event)
 
-    def persistAll[Event, State](events: im.Seq[Event]): Effect[Event, State] =
+    def persist[Event, State](event: Event, events: Event*): Effect[Event, State] =
+      if (events.nonEmpty)
+        Effect.persist(event :: events.toList)
+      else
+        new Persist[Event, State](event)
+
+    def persist[Event, State](eventOpt: Option[Event]): Effect[Event, State] =
+      Effect.persist[Event, State](eventOpt.toIndexedSeq)
```

Comment:
I'd rather pattern match here and return Effect.none or Persist(singleEvent)

Reply:
I was considering that as well, but then I was wondering what will happen if the callbacks when we switch to `Effect.none`. Do we apply callbacks when we explicitly say `Effect.none`? 

Do we (should) we allow this?

```scala
Effect
  .none
  .andThen ( ... )
```

If yes, we can map `None` to `Effect.none`. Otherwise, we should keep it as an empty `Seq`. An empty `Option` is equivalent to an empty `Seq` and in both cases, we should run the side-effects.

Reply:
Hmm, I think now that it should be legal to call
`Effect.none.andThen ( ... )`

Because we should also be able to call
`Effect.none.andThenStop()`

Bottom line, any register side-effect is unrelated to the fact of events being persisted or not.

Reply:
my point is that single event persist should not be turned into a PersistAll, we might have different optimizations for persist(singleEvent) and persistAll further down and the knowledge about that it was a persist(singleEvent) shouldn't be lost here in the upper layers

For `Option[Event]` it's know that it's one or no event.